File: dashboard/DiscordBot.Dashboard/src/app/core/models/command-panel.models.ts
export interface CommandPanelButton {
  id: string;
  action: string;
  label: string;
  style: 'Primary' | 'Secondary' | 'Success' | 'Danger';
  enabled: boolean;
  order: number;
}

export const COMMAND_PANEL_ACTIONS: { value: string; labelKey: string }[] = [
  { value: 'ticket_open', labelKey: 'settings.panel.actions.ticketOpen' },
  { value: 'ticket_help', labelKey: 'settings.panel.actions.ticketHelp' },
  { value: 'ping', labelKey: 'settings.panel.actions.ping' },
  { value: 'server_info', labelKey: 'settings.panel.actions.serverInfo' },
  { value: 'moderation_help', labelKey: 'settings.panel.actions.moderationHelp' },
  { value: 'reaction_roles_help', labelKey: 'settings.panel.actions.reactionRolesHelp' },
  { value: 'platform_help', labelKey: 'settings.panel.actions.platformHelp' }
];

export const COMMAND_PANEL_STYLES: { value: CommandPanelButton['style']; labelKey: string }[] = [
  { value: 'Primary', labelKey: 'settings.panel.styles.primary' },
  { value: 'Secondary', labelKey: 'settings.panel.styles.secondary' },
  { value: 'Success', labelKey: 'settings.panel.styles.success' },
  { value: 'Danger', labelKey: 'settings.panel.styles.danger' }
];

export const DEFAULT_COMMAND_PANEL_BUTTONS: CommandPanelButton[] = [
  {
    id: 'ticket-open',
    action: 'ticket_open',
    label: 'Create Ticket',
    style: 'Success',
    enabled: true,
    order: 0
  },
  {
    id: 'ticket-help',
    action: 'ticket_help',
    label: 'Ticket Help',
    style: 'Secondary',
    enabled: true,
    order: 1
  }
];

export type PanelButtonStyle = 'Primary' | 'Secondary' | 'Success' | 'Danger' | 'Link';
export type PanelButtonActionType = 'CreateTicket' | 'OpenUrl' | 'SendMessage' | 'AssignRole' | 'StartWorkflow';
export type PanelPublishStatus = 'NotPublished' | 'PendingPublish' | 'Published' | 'Failed';

export interface GuildPanelButton {
  id?: string; label: string; emoji?: string | null; style: PanelButtonStyle;
  actionType: PanelButtonActionType; ticketTypeId?: string | null; url?: string | null;
  workflowId?: string | null;
  responseMessage?: string | null; roleDiscordId?: string | null; sortOrder: number; isEnabled: boolean;
}

export interface GuildPanel {
  id: string; guildId: string; name: string; title: string; description: string; imageUrl?: string | null;
  channelDiscordId: string; messageDiscordId?: string | null; isEnabled: boolean; isPublished: boolean;
  refreshRequested: boolean; lastPublishedAtUtc?: string | null; createdAtUtc: string; updatedAtUtc: string;
  publishStatus: PanelPublishStatus; lastPublishFailed: boolean; lastPublishFailureReason?: string | null;
  lastPublishAttemptedAtUtc?: string | null;
  buttons: GuildPanelButton[];
}

export type SaveGuildPanel = Pick<GuildPanel, 'name' | 'title' | 'description' | 'imageUrl' | 'channelDiscordId' | 'isEnabled' | 'buttons'>;

File: src/DiscordBot.Infrastructure/Models/CommandPanelModels.cs
namespace DiscordBot.Infrastructure.Models;

using DiscordBot.Domain.Enums;

public sealed class CommandPanelButtonDefinition
{
    public string Id { get; set; } = string.Empty;
    public string Action { get; set; } = string.Empty;
    public string Label { get; set; } = string.Empty;
    public string Style { get; set; } = "Secondary";
    public bool Enabled { get; set; } = true;
    public int Order { get; set; }
}

public sealed class CommandPanelConfigDto
{
    public bool Enabled { get; init; }
    public string? ChannelId { get; init; }
    public string? MessageId { get; init; }
    public string Title { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public string? ImageUrl { get; init; }
    public IReadOnlyList<CommandPanelButtonDefinition> Buttons { get; init; } = [];
}

public sealed class CommandPanelRefreshDto
{
    public Guid PanelId { get; init; }
    public string DiscordGuildId { get; init; } = string.Empty;
    public string ChannelDiscordId { get; init; } = string.Empty;
    public string? MessageDiscordId { get; init; }
    public string Title { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public string? ImageUrl { get; init; }
    public IReadOnlyList<GuildPanelButtonDto> Buttons { get; init; } = [];
}

public sealed class AckCommandPanelRequest
{
    public string? MessageDiscordId { get; set; }
    public bool Success { get; set; }
    public string? FailureReason { get; set; }
}

public sealed class GuildPanelDto
{
    public Guid Id { get; init; }
    public Guid GuildId { get; init; }
    public string Name { get; init; } = string.Empty;
    public string Title { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public string? ImageUrl { get; init; }
    public string ChannelDiscordId { get; init; } = string.Empty;
    public string? MessageDiscordId { get; init; }
    public bool IsEnabled { get; init; }
    public bool IsPublished { get; init; }
    public bool RefreshRequested { get; init; }
    public PanelPublishStatus PublishStatus { get; init; }
    public DateTimeOffset? LastPublishedAtUtc { get; init; }
    public bool LastPublishFailed { get; init; }
    public string? LastPublishFailureReason { get; init; }
    public DateTimeOffset? LastPublishAttemptedAtUtc { get; init; }
    public DateTimeOffset CreatedAtUtc { get; init; }
    public DateTimeOffset UpdatedAtUtc { get; init; }
    public IReadOnlyList<GuildPanelButtonDto> Buttons { get; init; } = [];
}

public sealed class GuildPanelButtonDto
{
    public Guid Id { get; init; }
    public string Label { get; init; } = string.Empty;
    public string? Emoji { get; init; }
    public PanelButtonStyle Style { get; init; }
    public PanelButtonActionType ActionType { get; init; }
    public Guid? TicketTypeId { get; init; }
    public string? Url { get; init; }
    public string? ResponseMessage { get; init; }
    public string? RoleDiscordId { get; init; }
    public int SortOrder { get; init; }
    public bool IsEnabled { get; init; }
}

public sealed class SaveGuildPanelRequest
{
    public string Name { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? ImageUrl { get; set; }
    public string ChannelDiscordId { get; set; } = string.Empty;
    public bool IsEnabled { get; set; } = true;
    public List<SaveGuildPanelButtonRequest> Buttons { get; set; } = [];
}

public sealed class SaveGuildPanelButtonRequest
{
    public Guid? Id { get; set; }
    public string Label { get; set; } = string.Empty;
    public string? Emoji { get; set; }
    public PanelButtonStyle Style { get; set; }
    public PanelButtonActionType ActionType { get; set; }
    public Guid? TicketTypeId { get; set; }
    public string? Url { get; set; }
    public string? ResponseMessage { get; set; }
    public string? RoleDiscordId { get; set; }
    public int SortOrder { get; set; }
    public bool IsEnabled { get; set; } = true;
}

public sealed class PanelButtonActionDto
{
    public string DiscordGuildId { get; init; } = string.Empty;
    public Guid PanelId { get; init; }
    public Guid ButtonId { get; init; }
    public PanelButtonActionType ActionType { get; init; }
    public Guid? TicketTypeId { get; init; }
    public string? Url { get; init; }
    public string? ResponseMessage { get; init; }
    public string? RoleDiscordId { get; init; }
}

public sealed class TicketChannelCleanupDto
{
    public Guid TicketId { get; init; }
    public Guid GuildId { get; init; }
    public string DiscordGuildId { get; init; } = string.Empty;
    public string ChannelDiscordId { get; init; } = string.Empty;
    public int TicketNumber { get; init; }
    public string OwnerDiscordUserId { get; init; } = string.Empty;
    public string? OwnerDisplayName { get; init; }
    public string? ClosedByDiscordUserId { get; init; }
    public string? ClosedByDisplayName { get; init; }
    public DateTimeOffset? ClosedAt { get; init; }
    public string? TicketArchiveChannelId { get; init; }
    public string TicketClosedFromDashboardMessage { get; init; } = string.Empty;
}
